### PR TITLE
fix(profile): cover extraction and consolidation timestamp paths

### PR DESCRIPTION
**Pull request by joshuaswarren**
Part of #2134

Covers the deferred profile-header boundary findings from #2164 and adds integration coverage for extraction persistence and smart consolidation.

Verification:
- pnpm --filter @remnic/core exec tsx --test src/storage-profile-timestamp.test.ts
- pnpm exec tsx --test tests/profile-header-pipeline.test.ts
- pnpm --filter @remnic/core run check-types
- pnpm exec tsc --noEmit --project tsconfig.json

---

> [!NOTE]
> **Low Risk**
> Changes are localized to profile markdown parsing and test coverage; behavior shifts only for NBSP link refs and HTML-after-indented-code edge cases, with no auth or persistence contract changes beyond header placement.
> 
> **Overview**
> Tightens **profile `*Last updated*` header** placement in `profile-header.ts` so edge-case Markdown is handled correctly, with matching unit tests and new **orchestrator pipeline** coverage.
> 
> **Link-reference boundaries** now require ASCII space or tab after `]:` (not arbitrary whitespace). Lines like `[docs]: /url` with a non-breaking space are no longer treated as reference definitions, so a following stale header is not refreshed in the wrong place.
> 
> **Generic HTML blocks** may start immediately after a **completed indented code block**, keeping following custom tags and embedded `*Last updated*` lines opaque during metadata scanning instead of mis-identifying headers inside HTML.
> 
> Adds **`tests/profile-header-pipeline.test.ts`** to assert that **`persistExtraction`** and **`runConsolidationNow`** refresh `profile.md`’s timestamp before appending profile updates when the file already has a stale header.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 1ff86a406ff4ad7a1a099579d35f2a03ab376532. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved profile timestamp refresh handling when profiles contain HTML, indented code blocks, or link references.
  * Preserved embedded HTML and link-reference content while updating stale “Last updated” headers.
  * Ensured profile headers refresh correctly when saving extraction results or consolidated updates.

### CODE REVIEW

File: tests/profile-header-pipeline.test.ts [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+      [
+        "# Behavioral Profile",
+        "",
+        FRESH_HEADER,
+        "",
+        "- Existing profile note.",
+        "- Prefers concise updates.",
+        "",
+      ].join("\n"),
+    );
```
[the remark this answers is not in the document]
Comment (joshuaswarren) [reply tests/profile-header-pipeline.test.ts:83]: Thanks; I evaluated this suggestion. The tests intentionally assert the end-to-end persisted profile contract. Refresh-versus-append call order is an internal implementation detail, and instrumenting those calls would couple the tests to method order without adding an observable contract. The production path already refreshes through the centralized profile writer before appending, and the focused profile tests cover header freshness. Keeping call-order instrumentation out of scope for #2178.
